KOService.Presentation/KOService/src/app/mechanic/home/activities/activities.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import { ActivityService } from 'src/app/shared/services/activity.service';
+import { trigger, state, style, transition, animate } from '@angular/animations';
+import { Activity } from 'src/app/shared/models/Activity';
+import { ActivityStatus } from 'src/app/shared/enums/ActivityStatus';
+import { Repair } from 'src/app/shared/models/Repair';
+import { RepairService } from 'src/app/shared/services/repair.service';
+
+@Component({
+  selector: 'app-activities',
+  templateUrl: './activities.component.html',
+  styleUrls: ['./activities.component.css'],
+  animations: [
+    trigger('detailExpand', [
+      state('collapsed', style({height: '0px', minHeight: '0', display: 'none'})),
+      state('expanded', style({height: '*'})),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
+    ]),
+  ],
+})
+
+export class ActivitiesComponent implements OnInit {
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+  dataSource = new MatTableDataSource<Activity>();
+  repairs: Repair[];
+  userId: string;
+  displayedColumns = ['requestTime', 'description', 'status', 'sequenceNumber','carNumbers', 'carBrand'];
+  columnsToDisplayMap = [  {name: 'requestTime', display: 'data'},
+    {name: 'description', display: 'opis'}, 
+    {name: 'status', display: 'status'},
+    {name: 'sequenceNumber', display: 'poziom ważności'
+  }];
+
+  constructor(
+    private activityService: ActivityService, 
+    private  repairService: RepairService
+  ){ }
+
+  ngOnInit() {
+    this.userId = localStorage.getItem('auth_key');
```

Comment:
widzę że nie korzystasz z tego na razie, ale tak na przyszłość w 'auth_key' jest cały token, a nie id usera, więc jeszcze musimy ustalić jak będziemy pobierali to id. Na razie na masterze w AuthService jest property które się nazywa currentEmployee i on ma id które nam będzie potrzebne, z tym że trzeba uważać na przeładowania strony, bo wtedy serwis tworzy się zawsze na nowo i przez chwile to pole jest nullem